File: Remove Palindromic Subsequences/removePalindromicSubsequences.js
/**
 * @param {string} s
 * @return {number}
 */
 function removePalindromeSub(str) {
    if (!str) return 0
    for (let i = 0, j = str.length - 1; i < j; i++, j--)
        if (str.charAt(i) !== str.charAt(j)) return 2
    return 1
};

/* 
  console.log(removePalindromeSub("ababa")) // 1
  console.log(removePalindromeSub("abb")) // 2
  */

File: Remove Palindromic Subsequences/removePalindromicSubsequences.java
class Solution {
    public int removePalindromeSub(String s) {
        if(s.isBlank()) return 0;
        
        if(isPalindrome(s)){
            return 1;
        }
        return 2;
    }
    
    boolean isPalindrome(String s) {
        int left=0, right=s.length()-1;
        
        while(left<right) {
            if(s.charAt(left)!=s.charAt(right)){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}

// Time complexity: O(N)
// Space complexity: O(1)
